templates/configmaps_st2-urls.yaml

```
@@ -7,6 +7,18 @@ metadata:
     description: StackStorm service URLs, used across entire st2 cluster
   labels: {{- include "stackstorm-ha.labels" (list $ "st2") | nindent 4 }}
 data:
+  {{- if and .Values.st2.tls.enabled .Values.st2auth.tls.enabled }}
+  ST2_AUTH_URL: https://{{ .Release.Name }}-st2auth:9100/
+  {{- else }}
   ST2_AUTH_URL: http://{{ .Release.Name }}-st2auth:9100/
```

We can reduce some of the repetition here by front loading the calculation:
```suggestion
  {{- if and .Values.st2.tls.enabled .Values.st2auth.tls.enabled }}
    {{- $proto := "https" }}
  {{- else }}
    {{- $proto := "http" }}
  {{- end }}
  ST2_AUTH_URL: {{ $proto }}://{{ .Release.Name }}-st2auth:9100/
```
and the same for the others.

---

oh. I see you're changing the ports for st2api and st2stream as well. Hmm.

st2auth already exposes the tls bits. Could we add tls to st2api and st2stream as well to avoid adding another process to the mix?

---

Sorry I don't follow - do you mean enhance the `st2apit` and `st2stream` libraries themselves to support TLS out-of-the-box instead of having to stick `stunnel` infront of them on the `k8s` end?

---

> Sorry I don't follow - do you mean enhance the `st2apit` and `st2stream` libraries themselves to support TLS out-of-the-box instead of having to stick `stunnel` infront of them on the `k8s` end?

Yes. `st2api` supports tls, so `st2api` and `st2stream` should do the same.

---

```suggestion
  {{- $proto := "http" }}
  {{- $port := "9100" }}
  {{- if and .Values.st2.tls.enabled .Values.st2auth.tls.enabled }}
    {{- $proto := "https" }}
    {{- $port := "9110" }}
  {{- end }}
  ST2_AUTH_URL: {{ $proto }}://{{ .Release.Name }}-st2auth:{{ $port }}/
```

---

So, it looks like we need to copy the st2auth ssl config opts to st2api and st2stream.

st2auth seems to define the ssl bits here:
- [st2auth/st2auth/config.py](https://github.com/StackStorm/st2/blob/180deff2f27f23da65e253a46f933365e40faf0f/st2auth/st2auth/config.py)
- [st2auth/st2auth/app.py](https://github.com/StackStorm/st2/blob/180deff2f27f23da65e253a46f933365e40faf0f/st2auth/st2auth/app.py)
- [st2auth/st2auth/cmd/api.py](https://github.com/StackStorm/st2/blob/180deff2f27f23da65e253a46f933365e40faf0f/st2auth/st2auth/cmd/api.py)

The SSL bits would need to go in these st2api files:
- [st2api/st2api/config.py](https://github.com/StackStorm/st2/blob/180deff2f27f23da65e253a46f933365e40faf0f/st2api/st2api/config.py)
- [st2api/st2api/app.py](https://github.com/StackStorm/st2/blob/180deff2f27f23da65e253a46f933365e40faf0f/st2api/st2api/app.py)
- [st2api/st2api/cmd/api.py](https://github.com/StackStorm/st2/blob/180deff2f27f23da65e253a46f933365e40faf0f/st2api/st2api/cmd/api.py)

The SSL bits would need to go in these st2stream files:
- [st2stream/st2stream/config.py](https://github.com/StackStorm/st2/blob/180deff2f27f23da65e253a46f933365e40faf0f/st2stream/st2stream/config.py)
- [st2stream/st2stream/app.py](https://github.com/StackStorm/st2/blob/180deff2f27f23da65e253a46f933365e40faf0f/st2stream/st2stream/app.py)
- [st2stream/st2stream/cmd/api.py](https://github.com/StackStorm/st2/blob/180deff2f27f23da65e253a46f933365e40faf0f/st2stream/st2stream/cmd/api.py)

---

Hey @cognifloyd I've opened https://github.com/StackStorm/st2/pull/6204 to add SSL/TLS support to `api` and `stream` if you could take a look that would be great :)